### ISSUE TITLE
Editing Tasks and To Dos - edit form closes immediately due to sync

### ISSUE THREAD

**Issue by Sambamm** [HabitRPG/habitica#3696]
Been noticing this for a while now - when I try to open and edit a Task or To Do, to add a check-list etc. it will automatically close 1 second later before I can change anything. I am using Google Chrome - I have tried FireFox and found the same issue.

**Comment by deilann** [comment]
It has to do if there are new notifs/changes since then. If you sync right beforehand, this should not happen.

**Comment by Sambamm** [comment]
I have tried that. I have also tried restarting the website, and my computer. I have also tried on a different computer. This problem is still happening.

**Comment by deilann** [comment]
It's a known issue and has to do with new notifications. That means that
you got another notification in the time between trying to sync and editing
the task. Restarting the site and your computer isn't going to help. Are
you in a lot of guilds with high activity? Sometimes when one of my guilds
is exploding, it will collapse every time I add a new checklist item.

**Comment by Sambamm** [comment]
I left the two challenges I was a part of and deleted the related tasks, and everything seems to be working fine now.
Hopefully in the future there will be a way to participate in Challenges and Guilds, and not have the items collapse mid-editing. But this will work for now; thanks!

**Comment by deilann** [comment]
Challenges should have nothing to do with it.

And I participate in many guilds -- the trick is just timing. I sync
immediately before starting editing. Once you've opened into edit mode,
wait for the blue bar to finish completely. If the blue bar finishes, the
task isn't going to collapse on you unless there's a new update next time
you trigger a save -- such as creating a new line on a checklist.

**Comment by Sambamm** [comment]
I'm not a member of any guilds - never have been - and the party I'm a part of is very quiet. Though as soon as I deleted the challenges, the problem was fixed. 
I added the challenges back, and regardless of sync (yes, I know how to properly do this with waiting for the blue bar to finish), the menu collapses immediately, even when I can guarantee there are no notifications.

**Comment by deilann** [comment]
Huh! This appears to be something different then.

Do you mind sharing which challenges they are?

**Comment by Sambamm** [comment]
Hence my confusion, and posts here haha :)

The challenges are: The Happy List Challenge, and Destroy the Drag'on!.

**Comment by deilann** [comment]
Oooh, those are both very large challenges, are they not?

Super large challenges do cause other problems... I wonder if this is
related.

**Comment by Sambamm** [comment]
It's very possible they're related - each has 40+ people in them. It's possible that whenever any of the members of the Challenge complete a task, habit or to-do, it causes the effects we've been seeing.

**Comment by deilann** [comment]
Mmm, 40+ is not that many. Not enough to trigger the kind of problems I'm
thinking about.

**Comment by Sambamm** [comment]
Did I say 40? I meant 372 and 175, respectively. I was totally looking at the wrong challenges... whoops :3

**Comment by SabreCat** [comment]
Maybe not when members hit tasks, but maybe when people join or drop out of the Challenge?

**Comment by Aiesila** [comment]
The closing-on-its-own problem when adding checklists to tasks shouldn't have anything to do with notifications/activities related to other users.  I'm 100% a habit hermit and it happens almost every time I add checklists to a to-do task.  
Checklist items sync with the server every time a new list slot is added or removed.  Other task changes don't appear to trigger a sync until the task is closed for editing via hitting save or cancel... or by hitting Enter, which then auto-saves/closes the editing options. I'd guess that having to hit Enter to add the next checklist slot is at least part of the issue.

**Comment by deilann** [comment]
No because often I will add a checklist item (press enter) and it's fine,
works as intended. And on the other hand sometimes I open the task to be
edited and the resulting sync collapses it.

There is something that it's drawing from a sync with the server that is
causing it to collapse. I thought it was notifications, but it may be
something else.

Are you in any challenges?

**Comment by Aiesila** [comment]
Nope. No challenges or guilds, and I'm a solo-quester.
The auto-close is definitely not consistent for me either.  Right now it won't do it at all! 

Such a mystery.  Please let me know if there is any other information I can provide.

**Comment by crookedneighbor** [comment]
I haven't seen this behavior, so I'm going to close this issue as it is quite old. If it's still occurring, please reopen.
